src/metadata/EntityMetadata.ts

```diff
@@ -570,7 +570,11 @@ export class EntityMetadata {
         if (!entity)
             return undefined;
 
-        return EntityMetadata.getValueMap(entity, this.primaryColumns, { skipNulls: true });
+        let fullEntity = entity;
+        if (this.discriminatorColumn && !entity.hasOwnProperty(this.discriminatorColumn.databasePath))
+            fullEntity[this.discriminatorColumn.databasePath] = this.discriminatorValue;
```

Comment:
Not sure I understand why you've assigned this variable and then are changing the prop in this variable.  Could you explain it to me?

If it's an object it's the same object in both variables - you haven't cloned it - so you might as well just `entity[this.discriminatorColumn.databasePath] = this.discriminatorValue;`